pallets/identity/src/lib.rs

```diff
@@ -249,8 +249,8 @@ decl_module! {
         const InitialPOLYX: <T::Balances as Currency<T::AccountId>>::Balance = T::InitialPOLYX::get().into();
 
         fn on_runtime_upgrade() -> Weight {
-            storage_migrate_on!(StorageVersion::get(), 1, {
-                migration::migrate_v1::<T>();
+            storage_migrate_on!(StorageVersion::get(), 2, {
+                migration::migrate_v2::<T>();
```

Comment:
Don't remove the v1 migration here.  Just duplicate the code to call your v2 migration.